dpp/src/document/state_transition/documents_batch_transition/validation/state/fetch_documents_factory.rs

```diff
@@ -0,0 +1,72 @@
+use futures::future::join_all;
+use serde_json::json;
+
+use crate::{
+    document::{document_transition::DocumentTransition, Document},
+    get_from_transition,
+    state_repository::StateRepositoryLike,
+    util::string_encoding::Encoding,
+};
+
+use std::collections::hash_map::{Entry, HashMap};
+
+pub struct FetchDocumentsFactory<SR>
```

Comment:
I'd probably rename it to something like `DocumentsFetcher` or `DocumentsRepository`